multicluster/controllers/multicluster/member/serviceexport_controller.go

```diff
@@ -231,19 +238,25 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			Namespace: req.Namespace,
 		},
 	}
-	readyAddresses := 0
-	err = r.Client.Get(ctx, req.NamespacedName, eps)
-	if err == nil {
-		for _, subsets := range eps.Subsets {
-			readyAddresses = readyAddresses + len(subsets.Addresses)
+
+	var hasReadyEndpoints bool
+	if r.endpointSliceEnabled {
+		eps.Subsets, hasReadyEndpoints, err = r.getSubsetsFromEndpointSlice(ctx, req)
+		if err != nil {
+			return ctrl.Result{}, err
 		}
 	} else {
-		if !apierrors.IsNotFound(err) {
+		hasReadyEndpoints, err = r.checkSubsetsFromEndpoint(ctx, req, eps)
+		if err != nil {
 			klog.ErrorS(err, "Failed to get Endpoints", req.String())
 			return ctrl.Result{}, err
 		}
 	}
-	if readyAddresses == 0 || apierrors.IsNotFound(err) {
+
+	if !hasReadyEndpoints {
+		// When the controller restarts, `svcInstalled` is false as the cache will be empty, but the available Endpoints of
+		// a Service might have been decreased to zero during the controller restarts, so we skip checking `svcInstalled`
```

Comment:
the controller restarts -> the controller restart

Reply:
Updated.